tests/test_command_macros.rs

```diff
@@ -0,0 +1,355 @@
+use crate::util::INTERNAL_SYMBOL_PREFIX;
+use egglog::ast::*;
+use egglog::util::FreshGen;
+use egglog::*;
+use egglog_ast::span::{RustSpan, Span};
+use std::sync::Arc;
+
+// Macro that prefixes rule names with a specific prefix
+struct PrefixRuleMacro {
+    prefix: String,
+}
+
+impl CommandMacro for PrefixRuleMacro {
+    fn transform(
+        &self,
+        command: Command,
+        symbol_gen: &mut util::SymbolGen,
+        _type_info: &TypeInfo,
+    ) -> Result<Vec<Command>, Error> {
+        match command {
+            Command::Rule { mut rule } => {
+                rule.name = symbol_gen.fresh(&format!("{}_{}", self.prefix, rule.name));
+                Ok(vec![Command::Rule { rule }])
+            }
+            cmd => Ok(vec![cmd]),
+        }
+    }
+}
+
+// Macro that duplicates every rule
+struct DuplicateRuleMacro;
+
+impl CommandMacro for DuplicateRuleMacro {
+    fn transform(
+        &self,
+        command: Command,
+        symbol_gen: &mut util::SymbolGen,
+        _type_info: &TypeInfo,
+    ) -> Result<Vec<Command>, Error> {
+        match command {
+            Command::Rule { rule } => {
+                let mut rule1 = rule.clone();
+                let mut rule2 = rule;
+                rule1.name = symbol_gen.fresh(&format!("dup1_{}", rule1.name));
+                rule2.name = symbol_gen.fresh(&format!("dup2_{}", rule2.name));
+                Ok(vec![
+                    Command::Rule { rule: rule1 },
+                    Command::Rule { rule: rule2 },
+                ])
+            }
+            cmd => Ok(vec![cmd]),
+        }
+    }
+}
+
+// Macro that adds a comment after each rule showing its name
+struct CommentAfterRuleMacro;
+
+impl CommandMacro for CommentAfterRuleMacro {
+    fn transform(
+        &self,
+        command: Command,
+        _symbol_gen: &mut util::SymbolGen,
+        _type_info: &TypeInfo,
+    ) -> Result<Vec<Command>, Error> {
+        match command {
+            Command::Rule { rule } => {
+                let rule_name = rule.name.clone();
+                Ok(vec![
+                    Command::Rule { rule },
+                    // Comments don't exist in the AST but we can use PrintSize with None
+                    // as a marker that the macro ran
+                    Command::PrintSize(span!(), Some(format!("rule_{}", rule_name))),
```

Comment:
Consider rewriting this example macro? It does not do what its name suggests

Reply:
done